spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java

```diff
@@ -373,4 +386,371 @@ public void testNotIn() {
+
+    Predicate notEq1 = new Predicate("<>", attrAndValue);
+    Expression expectedNotEq1 = Expressions.notEqual(item, value);
+    Expression actualNotEq1 = SparkV2Filters.convert(notEq1);
+    Assertions.assertThat(actualNotEq1.toString()).isEqualTo(expectedNotEq1.toString());
+
+    Predicate notEq2 = new Predicate("<>", valueAndAttr);
+    Expression expectedNotEq2 = Expressions.notEqual(item, value);
+    Expression actualNotEq2 = SparkV2Filters.convert(notEq2);
+    Assertions.assertThat(actualNotEq2.toString()).isEqualTo(expectedNotEq2.toString());
+
+    Predicate eqNullSafe1 = new Predicate("<=>", attrAndValue);
+    Expression expectedEqNullSafe1 = Expressions.equal(item, value);
+    Expression actualEqNullSafe1 = SparkV2Filters.convert(eqNullSafe1);
+    Assertions.assertThat(actualEqNullSafe1.toString()).isEqualTo(expectedEqNullSafe1.toString());
+
+    Predicate eqNullSafe2 = new Predicate("<=>", valueAndAttr);
+    Expression expectedEqNullSafe2 = Expressions.equal(item, value);
+    Expression actualEqNullSafe2 = SparkV2Filters.convert(eqNullSafe2);
+    Assertions.assertThat(actualEqNullSafe2.toString()).isEqualTo(expectedEqNullSafe2.toString());
+
+    Predicate in = new Predicate("IN", attrAndValue);
+    Expression expectedIn = Expressions.in(item, value);
+    Expression actualIn = SparkV2Filters.convert(in);
+    Assertions.assertThat(actualIn.toString()).isEqualTo(expectedIn.toString());
+
+    Predicate and = new And(lt1, eq1);
+    Expression expectedAnd = Expressions.and(expectedLt1, expectedEq1);
+    Expression actualAnd = SparkV2Filters.convert(and);
+    Assertions.assertThat(actualAnd.toString()).isEqualTo(expectedAnd.toString());
+
+    org.apache.spark.sql.connector.expressions.Expression[] attrAndAttr =
+        expressions(expression, expression);
+    Predicate invalid = new Predicate("<", attrAndAttr);
+    Predicate andWithInvalidLeft = new And(invalid, eq1);
+    Expression convertedAnd = SparkV2Filters.convert(andWithInvalidLeft);
+    Assertions.assertThat(convertedAnd).isNull();
+
+    Predicate or = new Or(lt1, eq1);
+    Expression expectedOr = Expressions.or(expectedLt1, expectedEq1);
+    Expression actualOr = SparkV2Filters.convert(or);
+    Assertions.assertThat(actualOr.toString()).isEqualTo(expectedOr.toString());
+
+    Predicate orWithInvalidLeft = new Or(invalid, eq1);
+    Expression convertedOr = SparkV2Filters.convert(orWithInvalidLeft);
+    Assertions.assertThat(convertedOr).isNull();
+
+    Predicate not = new Not(lt1);
+    Expression expectedNot = Expressions.not(expectedLt1);
+    Expression actualNot = SparkV2Filters.convert(not);
+    Assertions.assertThat(actualNot.toString()).isEqualTo(expectedNot.toString());
+  }
+
+  private org.apache.spark.sql.connector.expressions.Expression[] expressions(
+      org.apache.spark.sql.connector.expressions.Expression... expressions) {
+    return expressions;
+  }
+
+  private static int dateToYears(String dateString) {
```

Comment:
Why are there so many conversion functions here? There are conversion functions in the new helper as well. Are they all used?